pkg/provider/k8s/k8s.go

```diff
@@ -416,7 +433,49 @@ func (c *K8s) deploymentApply(resource runtime.Object) error {
 		func() (bool, error) { return c.deploymentReady(resource) })
 }
 
+func (c *K8s) customResourceDefinitionApply(resource runtime.Object) error {
+	req := resource.(*apiserverextensionsv1beta1.CustomResourceDefinition)
+	kind := resource.GetObjectKind().GroupVersionKind().Kind
+	if len(req.Namespace) == 0 {
+		req.Namespace = "default"
+	}
+
+	switch v := resource.GetObjectKind().GroupVersionKind().Version; v {
+	case "v1beta1":
+		client := c.apiextclt.ApiextensionsV1beta1().CustomResourceDefinitions()
+		list, err := client.List(apiMetaV1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "error listing resource : %v, name: %v", kind, req.Name)
+		}
+		var exists bool
+		for _, l := range list.Items {
+			if l.Name == req.Name {
+				exists = true
+				break
+			}
+		}
+		if exists {
+			if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				_, err := client.Update(req)
+				return err
+			}); err != nil {
+				return errors.Wrapf(err, "resource update failed - kind: %v, name: %v", kind, req.Name)
+			}
+			log.Printf("resource updated - kind: %v, name: %v", kind, req.Name)
+			return nil
+		} else if _, err := client.Create(req); err != nil {
+			return errors.Wrapf(err, "resource creation failed - kind: %v, name: %v", kind, req.Name)
+		}
+		log.Printf("resource created - kind: %v, name: %v", kind, req.Name)
+	default:
+		return fmt.Errorf("unknown object version: %v kind:'%v', name:'%v'", v, kind, req.Name)
+	}
+
+	return nil
+}
+
 func (c *K8s) ingressApply(resource runtime.Object) error {
+	// TODO(geekodour): maybe use Ingress from networking.k8s.io/v1beta1
```

Comment:
Instead if this comment wouldn't be better to open a new issue and adding more details?

Reply:
I looked into a bit more, this will be required when we upgrade k8s version to 1.14.x so we can leave it as is now. I'll remove the comment.